### PR TITLE
Add support for SUSE 15, Proactive distro check

### PR DESCRIPTION
**Pull request by hestolz**
Currently OMS extension does not support SUSE 15.
- Added the distro to the allowed list as well as the list of tested distros.
- For above, manually parse OS release because python .dist() does not do so properly
- In test automation, precheck existence of desired VM images and fail immediately if not all are found

### CODE REVIEW

File: OmsAgent/extension-test/omsfiles/oms_extension_run_script.py [hunk 1]
```diff
@@ -80,13 +80,25 @@ def is_vm_supported_for_extension():
...
         vm_dist, vm_ver, vm_id = platform.dist()
 
+    if not vm_dist and not vm_ver: # SLES 15
+        with open('/etc/os-release', 'r') as fp:
+            for line in fp:
+                if line.startswith('ID='):
+                    name = line.split('=')[1]
+                    name = name.split('-')[0]
+                    name = name.replace('\"', '').replace('\n', '')
+            elif line.startswith('VERSION_ID='):
```
Comment (AaronRM) [inline OmsAgent/extension-test/omsfiles/oms_extension_run_script.py]: Indentation needs to be fixed.

File: OmsAgent/omsagent.py [hunk 2]
```diff
@@ -621,14 +621,26 @@ def is_vm_supported_for_extension():
...
         vm_dist, vm_ver, vm_id = platform.dist()
 
+    if not vm_dist and not vm_ver: # SLES 15
+        with open('/etc/os-release', 'r') as fp:
+            for line in fp:
+                if line.startswith('ID='):
+                    name = line.split('=')[1]
+                    name = name.split('-')[0]
+                    name = name.replace('\"', '').replace('\n', '')
+            elif line.startswith('VERSION_ID='):
```
Comment (AaronRM) [inline OmsAgent/omsagent.py]: Indentation needs to be fixed.

File: OmsAgent/extension-test/omsfiles/oms_extension_run_script.py (hunk 1 above)
Comment (hestolz) [reply OmsAgent/extension-test/omsfiles/oms_extension_run_script.py]: edited

File: OmsAgent/extension-test/README.md [hunk 3]
```diff
@@ -79,8 +79,6 @@ $ pip install requests adal json2html rstr
   - `<nsg resource group>` – resource group of your NSG
   - `<nsg>` – NSG name
   - `<size>` – Standard_B1ms
-  - `<publisher>` - pulisher name (For OMSAgent: Microsoft.EnterpriseCloud.Monitoring)
```
Comment (hestolz) [inline OmsAgent/extension-test/README.md]: removing since always same

File: OmsAgent/extension-test/omsfiles/oms_extension_run_script.py [hunk 4]
```diff
@@ -80,13 +80,25 @@ def is_vm_supported_for_extension():
...
+            for line in fp:
+                if line.startswith('ID='):
+                    vm_dist = line.split('=')[1]
+                    vm_dist = vm_dist.split('-')[0]
+                    vm_dist = vm_dist.replace('\"', '').replace('\n', '')
+                elif line.startswith('VERSION_ID='):
+                    vm_ver = line.split('=')[1]
+                    vm_ver = vm_ver.split('.')[0]
+                    vm_ver = vm_ver.replace('\"', '').replace('\n', '')
+
```
Comment (nidhanda) [inline OmsAgent/extension-test/omsfiles/oms_extension_run_script.py]: Why do we need to do this specifically for SLES 15?

Comment (hestolz) [reply OmsAgent/extension-test/omsfiles/oms_extension_run_script.py]: `platform.dist()` / `platform.linux_distribution()` returns empty strings on SLES 15 :|

**Pull request merged**